Update svm.rst. For the issue #20075

tom1092 opened the pull request:
There was a wrong formulation of Linear SVC. In the hinge loss term "1-" was missing.

Fixes https://github.com/scikit-learn/scikit-learn/issues/20075

tom1092 committed:
Update svm.rst. For the issue #20075

There was a wrong formulation of Linear SVC. In the hinge loss term "1-" was missing.

https://github.com/scikit-learn/scikit-learn/issues/20075

NicolasHug commented on doc/modules/svm.rst:
```diff
@@ -677,7 +677,7 @@ The primal problem can be equivalently formulated as
 .. math::
 
-    \min_ {w, b} \frac{1}{2} w^T w + C \sum_{i=1}\max(0, y_i (w^T \phi(x_i) + b)),
+    \min_ {w, b} \frac{1}{2} w^T w + C \sum_{i=1}\max(0, 1-y_i (w^T \phi(x_i) + b)),
```
The `-` should include the bias term as well:
(I'm not sure this will render nicely in latex, we should double check:)

```suggestion
    \min_ {w, b} \frac{1}{2} w^T w + C \sum_{i=1}\max(0, 1 - [ y_i (w^T \phi(x_i) + b] )),
```

tom1092 replied:
I think square brackets are not so relevant in this case, but I'll pr with this change.

tom1092 committed:
Updated svm.rst. Error in formulation of LinearSVC 

There was an error on the formulation of Linear SVC. The "1-" was missing in the hinge loss.

https://github.com/scikit-learn/scikit-learn/issues/20075

TomDLT reviewed (approved):
Before 
![image](https://user-images.githubusercontent.com/11065596/118338539-442b9600-b4cb-11eb-981a-45ebfe0d0bf3.png)
After 
![image](https://user-images.githubusercontent.com/11065596/118338558-4b52a400-b4cb-11eb-9128-213c653f6b82.png)

Not sure that the brackets are necessary, but thanks for the fix !

NicolasHug commented:
Yeah we don't need them, sorry I missed the inner parenthesis

glemaitre commented on doc/modules/svm.rst:
```diff
@@ -677,7 +677,7 @@ The primal problem can be equivalently formulated as
 .. math::
 
-    \min_ {w, b} \frac{1}{2} w^T w + C \sum_{i=1}\max(0, y_i (w^T \phi(x_i) + b)),
+    \min_ {w, b} \frac{1}{2} w^T w + C \sum_{i=1}\max(0, 1 - [y_i (w^T \phi(x_i) + b)]),
```
```suggestion
    \min_ {w, b} \frac{1}{2} w^T w + C \sum_{i=1}\max(0, 1 - y_i (w^T \phi(x_i) + b)),
```

glemaitre committed:
Update doc/modules/svm.rst

glemaitre commented:
Thanks @tom1092

tom1092 commented:
U’re welcome!

> Il giorno 17 mag 2021, alle ore 10:47, Guillaume Lemaitre ***@***.***> ha scritto:
> 
> ﻿
> Thanks @tom1092
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or unsubscribe.
